### PR TITLE
AI Generated Quizzes from Material URL

### PR DESCRIPTION
**Pull request by holoisme**
Hello,

This PR adds the ability to generate quizzes from material PDF documents.
The scope of this PR is only concerned with the back-end, so everything from PDF parsing, proper task prompt and raw AI answer parsing. The front-end by @FerdinandGomerChiu still needs to be able to receive it and will come in the following days.

### CODE REVIEW

File: edweiss-firebase/functions/src/functions/action/generateQuizContentFromMaterial.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		pdfContent = (await pdf(rawData)).text;
+	} catch (e) {
+		return fail("error-parsing-pdf");
+	}
+
+	if (pdfContent == undefined || pdfContent == null)
+		return fail("pdf-content-undefined");
+
+	/**
+questionmcq<TOKEN_END_QUESTION>
```
Comment (FerdinandGomerChiu) [inline edweiss-firebase/functions/src/functions/action/generateQuizContentFromMaterial.ts:28]: A nitpick: this comment can be removed.

**Comment by FerdinandGomerChiu** [comment]
I see you changed the toggleQuizResult cloud function, and added a new type in the lecture model. Have you made sure to not create conflicts with @adamm-alaoui-pro , as he is also working with those files?

File: edweiss-firebase/functions/src/functions/action/generateQuizContentFromMaterial.ts (hunk 1 above)
Comment (holoisme) [reply edweiss-firebase/functions/src/functions/action/generateQuizContentFromMaterial.ts:28]: I think it should instead be formally explicited as the "question" format for potential future refactors.

**Comment by holoisme** [comment]
> I see you changed the toggleQuizResult cloud function, and added a new type in the lecture model. Have you made sure to not create conflicts with @adamm-alaoui-pro , as he is also working with those files?

I didn't do much in those files, I just corrected his force-merge mistakes for me to compile my code.

File: edweiss-firebase/functions/src/functions/action/generateQuizContentFromMaterial.ts (hunk 1 above)
Comment (FerdinandGomerChiu) [reply edweiss-firebase/functions/src/functions/action/generateQuizContentFromMaterial.ts:28]: Ah, I see! Thanks for clearing up the confusion. UwU

**Review by FerdinandGomerChiu — APPROVED** [review]
This is insane work, especially considering you did it in about 30 min (after spending some time on flashcard generation). You are doing heavy lifting 🏋🏽. Keep it up!

**Pull request merged**